### PR TITLE
Add method to delete pages of the document by index

### PR DESCRIPTION
**Pull request by davidsilveira**
https://github.com/prawnpdf/prawn/issues/958

### CODE REVIEW

File: lib/prawn/document.rb [hunk 1]
```diff
@@ -306,6 +306,26 @@ def start_new_page(options = {})
...
+    # Remove the last page of the document
+    #
+    #   pdf = Prawn::Document.new
+    #   pdf.page_count #=> 1
+    #   3.times { pdf.start_new_page }
+    #   pdf.page_count #=> 4
+    #   pdf.delete_last_page
+    #   pdf.page_count #=> 3
+    #
+
```
Comment (sobrinho) [inline lib/prawn/document.rb]: empty line

File: lib/prawn/document.rb [hunk 2]
```diff
@@ -306,6 +306,26 @@ def start_new_page(options = {})
...
+    def delete_last_page
+      last_page = state.page
+      if last_page
+        state.pages.delete_at(state.pages.index(last_page))
+        state.store.pages.data[:Kids].pop
+        state.store.pages.data[:Count] -= 1
+        @page_number -= 1
+      end
```
Comment (sobrinho) [inline lib/prawn/document.rb]: Use earlier return:

```ruby
def delete_last_page
  last_page = state.page

  return if last_page.nil?

  ...
end
```

**Comment by sobrinho** [comment]
As we discussed, try to abstract to `delete_page(index)` to be able to delete any page instead of only the latest.

**Comment by pointlessone** [comment]
I wonder what is the use case for this feature.

**Comment by sobrinho** [comment]
See #958.

**Comment by pointlessone** [comment]
@sobrinho Thank you.

I'm not convinced that use case warrants an addition to the library.

**Comment by sobrinho** [comment]
@pointlessone this is a pattern that repeats over and over again in all projects that I work.

It's okay if you think that doesn't belong to the core but would be a good helper for us!

**Comment by pointlessone** [comment]
If you want each thing on its own page why don't you start a new page before rendering a thing?

**Comment by sobrinho** [comment]
Because you can't start a new page in the first iteration, otherwise the blank page would be first one instead of the latest.

In that case, the code is like that:

```ruby
things.each do |thing|
  if things.first != thiing
    document.start_new_page
  end

  render
end
```

**Comment by pointlessone** [comment]
I see but I'm still hesitant since there's a straight-forward workaround and the code is not complex or obscure.

**Comment by sobrinho** [comment]
Your call!

Maybe there is other use cases, ours is just this one.

**Comment by johnnyshields** [comment]
Deleting a page at an index is valid request. Ideally it should support negative indexes (`-1` == last page).

The use case given is rather weak/inelegant however.

**Comment by sobrinho** [comment]
Awesome!

**Comment by pointlessone** [comment]
@davidsilveira Please make Rubocop happy. Otherwise it's good.

**Pull request merged**